api/python/provisioner/inputs.py

```diff
@@ -612,13 +612,42 @@ class NTP(ParamGroupInputBase):
 @attr.s(auto_attribs=True)
 class Hostname(ParamGroupInputBase):
     _param_group = 'hostname'
+    hostname: str = ParamGroupInputBase._attr_ib(
+        _param_group, descr="hostname to be set"
+    )
 
 
 @attr.s(auto_attribs=True)
 class Firewall(ParamGroupInputBase):
     _param_group = 'firewall'
 
 
+@attr.s(auto_attribs=True)
+class DataNetwork(ParamGroupInputBase):
+    _param_group = 'data_network'
+    data_public_ip: str = ParamGroupInputBase._attr_ib(
+        _param_group, descr="node public data interface IP",
+        validator=Validation.check_ip4
+    )
+    data_gateway: str = ParamGroupInputBase._attr_ib(
```

Comment:
This is optional parameter. Would validation fail if it's blank?

Reply:
Yes, It will skip validation if value is missing
https://github.com/Seagate/cortx-prvsnr/blob/aeb708e705abd86731b69c2f1898a7a340eafefb/api/python/provisioner/inputs.py#L576